src/services/helper.ts

```diff
@@ -52,7 +34,7 @@ export class HelperService<T extends ChainId> extends ContractService<T> {
    * @returns list of erc20 object
    */
   async tokens(addresses: Address[], overrides: CallOverrides = {}): Promise<ERC20[]> {
-    return await this.contract.read.tokensMetadata(addresses, overrides).then(structArray);
+    return await this.contract.then(contract => contract.read.tokensMetadata(addresses, overrides)).then(structArray);
```

Comment:
We are missing the refactor of defining `const contract = await this.contract;` on this file